experimental/packages/opentelemetry-instrumentation-http/README.md

```diff
@@ -57,6 +57,8 @@ Http instrumentation has few options available to choose from. You can set the f
 | [`startOutgoingSpanHook`](https://github.com/open-telemetry/opentelemetry-js/blob/main/experimental/packages/opentelemetry-instrumentation-http/src/types.ts#L99) | `StartOutgoingSpanCustomAttributeFunction` | Function for adding custom attributes before a span is started in outgoingRequest |
 | `ignoreIncomingRequestHook` | `IgnoreIncomingRequestFunction` | Http instrumentation will not trace all incoming requests that matched with custom function |
 | `ignoreOutgoingRequestHook` | `IgnoreOutgoingRequestFunction` | Http instrumentation will not trace all outgoing requests that matched with custom function |
+| `instrumentOutgoingRequests` | `boolean` | Set to false to avoid instrumenting outgoing requests at all. |
+| `instrumentIncomingRequests` | `boolean` | Set to false to avoid instrumenting incoming requests at all. |
```

Comment:
Some issues here:
1. Generally the JS SDK, at least, tries to use config vars that are false-y by default. This is because then a `false` value is treated the same as an `undefined` value. The `<Instrumentation>._config` object on the various instrumentations in this and the opentelemetry-js-contrib repo are often just a sparse object where undefined config vars implies the default value. (The huge exception here is the `enabled: true` config var on all instrumentations. :sadface:) It would potentially be error prone to have a config var where `false` and `undefined` are different behaviours.
2. From the config vars and documentation, it isn't clear which of `instrumentOutgoingRequests` and `ignoreOutgoingRequestHook` should take precedence.


Without complicating this PR by getting too general, one possible option:
- Change the verb from "instrument" to "disable", e.g. `disableIncomingRequestInstrumentation`. This flips the boolean sense to deal with issue 1. The term "disable" is closer to the enable/disable state on the an entire Instrumentation, which helps imply that this take precedence over the `ignore*Hook` vars -- though the docs should still spell that out.

What do you think?



3. It isn't clear from the docs here that `instrument*` vars are just skipping this one span and the `ignore*` are *supressing*. What exactly the existing `ignore*RequestHook` vars were doing already wasn't clear -- which is an existing issue.  Even the term "suppress" here isn't clear ... given that the OTel Java SDK is using (https://opentelemetry.io/docs/zero-code/java/agent/configuration/#instrumentation-span-suppression-behavior) "suppression" to mean just dropping a span and not suppressing the full trace.

- I think this could be solved by expanding on the description (or in a `[1] ...`-note below the options table) in this section of the README. Personally, I think mentioning the particular use case could be helpful for other users to understand why they might want to use this, and to inform a possible future generalization.

Reply:
Changing the name to `disableIncomingRequestInstrumentation` makes a lot of sense to me, that's def. better! I'll update the PR accordingly. I'll also update the docs part to explain this better, thanks for the feedback!